components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -8657,6 +8672,59 @@ public boolean isFavouriteApp(APIIdentifier identifier, String username, int ten
         return apiIdentifiers;
     }
 
+    private void getUserAccessibleAppsByBusinessOwner(List<APIIdentifier> apiIdentifiers, Map<String, List<String>>
+            businessOwnerIdsMap, Registry registry, int tenantId, String userName) throws AppManagementException {
+        boolean isTenantFlowStarted = false;
+        try {
+            UserRealmService realmService =
+                    (UserRealmService) PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                            .getOSGiService(UserRealmService.class);
+            String requestedTenantDomain = realmService.getTenantManager().getDomain(tenantId);
+
+            if (requestedTenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(
+                    requestedTenantDomain)) {
+                isTenantFlowStarted = true;
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(requestedTenantDomain, true);
+            }
+
+            APIInfoDTO[] subscribedApps = getSubscribedAPIsOfUser(userName);
+            // User has to set anonnymous to get myapps.
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(CarbonConstants
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/4642888b8ad6934dcda162142607b8549b0728e5